Zend/zend_types.h

```diff
@@ -331,6 +336,24 @@ struct _zval_struct {
 	} u2;
 };
 
+typedef struct _zend_refcounted_h {
```

Comment:
`zend_refcounted.h` should be probably kept and as included in original `rend_types.h`, there should be no BC break

Reply:
No, as the RFC was declined.

Reply:
The RFC was introducing some BC breaking changes. This one is not BC breaking. So is there any strong rationale to revert `zend_refcounted.h`?

Reply:
Yes. "Include splitting" was an additional voting question of the RFC and the whole RFC was declined.

Reply:
The main "Should #include directives be cleaned up?" RFC question is too broad to be taken seriously. "Is it allowed to split a large header to reduce dependencies?" would actually pass if the main RFC question would pass. Based on my knowledge, a minor question integrated in one declined RFC has no power. This was disucssed in the maining list in the past several times with consensus.

My point on `zend_refcounted.h` is it is not BC-breaking and I agree with https://github.com/php/php-src/pull/11008#issuecomment-1495209289 here. As I have written above, is there any strong, technical, rationale to revert it?

Reply:
https://wiki.php.net/rfc/voting#required_majority

Reply:
What do you want to say? IMHO too broad question cannot freeze changes for otherwise welcomed (minor/secondary) changes. IMHO the RFC process is not even needed for this change. So can we keep this discussion on technical level, please?

Reply:
As the author and the maintainer of the most parts of that code, I was against this refactoring at the first place.
As we didn't have a legal way to resolve that technical disagreements, I asked for voting.
The voting declined include clean-up.
Anyway, some commits were merged despite the common decision.
Some of that commits made troubles and were reverted quickly. The others were kept.
Now I want to revert the commits merged against the common decision.

Reply:
We will hopefully have soon a different process for handling technical changes (proposal for that coming soon) so there is no point to discuss it here. For now the RFC is the only process and it was decided against those changes so I think it should be reverted.